### DISCUSSION TITLE
Add recursive option to methods (or document an example)

### DISCUSSION THREAD

**Discussion by tacman** [ToshY/BunnyNet-PHP#129] [Q&A]
It appears that the underlying library does not support a recursive directory listing.  It'd be great to have that as an option.

Alternatively, is there example code to that this?  I know it's not that hard, but if you've already one it I'll happy use it!

Ideally, the recursive option would exist everywhere -- listing files, uploading a directory, downloading a directory (as a zip file), etc.

Plus, doesn't Flysystem have some sort of recursive option in its API?

I know this API is mostly a wrapper around the Bunny API calls, which don't support recursive (which would be even better), but certainly your library could regardless.  Alas, is potentially a LOT of api calls, but that's the nature of their API not supporting it.  

Amazon S3 supports recursive, and Symfony's Filesystem, etc. 

I'm working on a bundle which I'll mention in another discussion, and this is one of the things I ran into.  If you have some sample code for recursion, I can add it to the bundle until it's incorporated into your library.

Thx.

**Comment by ToshY** [comment]
Hey @tacman 

Is this question/discussion different from what eventually the conclusion was in #123 ? 

As currently there is no recursive option for the API, adding a recursive option/method doesn't seem the responsibility of this library. You've already showed the [snippet](https://github.com/ToshY/BunnyNet-PHP/discussions/123#discussioncomment-9689129) that the Flysystem adapter uses, so would think you can almost do the exact same with the API. If I have more time and can see if I can create a snippet to do this recursively. If it "works", I can add the example to the docs, but I'd rather not implement it in the library itself until it's officially supported.

**Comment by ToshY** [comment]
No longer active discussion; closed as duplicate of #123
